docs/development/near-clitool.md

```diff
@@ -22,7 +22,7 @@ You use NEAR Shell to connect to the NEAR platform from the terminal to create a
 To install NEAR Shell globally (recommended)
 
 ```bash
-npm install -g near-shell
+<snippet id='shell-install-command'/>
```

Comment:
This syntax feels strange to me. I also don't like that we have three places to look to fully understand what's happening here:

1. this file (and anywhere else that might use this syntax)
2. the place where we define that this snippet should expand to `npm install -g near-shell`
3. the place where we test that `npm install -g near-shell` behaves as we expect

If we are building the processor for these snippets ourselves, we have a lot of liberty with this syntax. How would you feel about something like this?

```markdown
To install NEAR Shell globally (recommended):

    npm install -g near-shell

<!-- TEST-SNIPPET
execSync('npm install -g near-shell')
const output = execSync('near state test.near')
assertEqual(output, 'whatever')
-->
```

Note here that I'm not automating consistency between the first line of the test and the actual command text, but since they're right next to each other it should be straightforward for humans to notice if they become out-of-sync. Also, if we didn't run this through a preprocessor, it would still look correct, since the test code is in a comment.

Then our preprocessor can look for comment blocks starting with `TEST-SNIPPET` and run them with `jest` or whatever, and include things like `execSync` so that our test snippets don't get cluttered. (I'm completely unconvinced of the `TEST-SNIPPET` identifier, or using jest, or using `execSync`. Feel free to suggest alternatives to all of these.)

Reply:
I think this is a lot harder to do. Our tests are going to be using diff technologies depending on what we are testing (e.g. near shell testing is best via a cli script, assembly script contract code is best using the unit testing tools we use in the examples, etc. It's very easy to do if we have explicit tests that reference snippets but very hard the other way around.

Reply:
Another feature we'll lose with defining tests in the .md file: we can no longer de-duplicate some content. Specifically for near-shell installation line, it features in several pages and will continue to be duplicated like this, except now possibly with dupe tests as well. If we continue with this PR, all the places that need a "how to install shell" instruction can just reference this content which is defined in one place.

Reply:
Ah, gotcha, that makes sense to me. Thanks for explaining and expanding the example!